katz/src/main/kotlin/katz/NonEmptyList.kt

```diff
@@ -66,6 +66,6 @@ class NonEmptyList<out A> private constructor(
     }
 
     companion object Factory {
-        fun <A> of(head: A, vararg t: A): NonEmptyList<A> = NonEmptyList(head, t.asList())
+        fun <A: Any> of(head: A, vararg t: A): NonEmptyList<A> = NonEmptyList(head, t.asList())
     }
```

Comment:
I thought the compiler could infer the restriction but then it is ok for me as is :)